src/modeler/AutoRest.Swagger/Validation/BodyTopLevelProperties.cs

```diff
@@ -38,11 +38,11 @@ public override bool IsValid( Dictionary<string, Operation> path, RuleContext co
                         {
                             if (param.Schema.Reference != null)
                             {
-                                string defName = param.Schema.Reference.Split('/').Last();
+                                string defName = Extensions.StripDefinitionPath(param.Schema.Reference);
                                 var definition = ((ServiceDefinition)context.Root).Definitions[defName];
                                 if (definition != null && definition.AllOf != null)
                                 {
-                                    if (definition.AllOf.Select(s => s.Reference).Where(reference => resourceRefRegEx.IsMatch(reference)) !=null)
+                                    if (definition.AllOf.Select(s => s.Reference).Where(reference => resourceRefRegEx.IsMatch(reference)) !=null && definition.Properties != null)
```

Comment:
Question:  would it be better to check for `definition.Properties != null` before making linq query to avoid unnecessary computation?